chapters/packages.tex

```diff
@@ -321,3 +321,148 @@ \section{External Resources}\label{external-resources}
 \filename{modelica://Modelica/Mechanics/C.jpg} is illegal.
 The references \filename{modelica://Modelica.Mechanics/C.jpg} and \filename{modelica://Modelica/C.jpg} must also refer to two distinct resources.
 \end{example}
+
+\section{Multilingual Descriptions}\label{multilingual-descriptions}
+
+\begin{nonnormative}
+Descriptive texts in a model or library are usually formulated in English.
+This section describes how a tool can present the library in another language.
+Translated Modelica text is provided by external files, so that no modification of the original Modelica text is required.
+\end{nonnormative}
+
+The texts in following Modelica constructs should be translated:
+\begin{itemize}
+\item description strings of component declarations and classes
+\item strings in the following annotations:
+\begin{itemize}
+\item \lstinline!Text.string!, \lstinline!Text.textString!
+\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
+\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
+\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
+\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
+\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
+\end{itemize}
+\end{itemize}
+
+Comments (delimited as well as rest-of-line) are not translated.
+Only literal strings and concatenations of literals strings are translated.
```

Comment:
Once more trying to get clarification about exactly what is supposed to be translated.
```suggestion
Where translation applies, the strings to be translated are the maximum subexpressions consisting of one or more concatenated string literals.

\begin{example}
Consider the following \lstinline!Text.string! (which is a construct where translation applies):
\begin{lstlisting}[language=modelica]
annotation(…, Text(string = "Hello, " +
    "the " + getInstanceName() +
    " is an instance of %class."), …);
\end{lstlisting}
The two strings to be translated are:
\begin{itemize}
\item \emph{Hello, the\ }
\item \emph{\ is an instance of \%class.}
\end{itemize}
\end{example}

```

Again, one of my main reasons for being able to translate sub-expressions is that it is common to have `sprintf`-like functions, where the string to be translated is one of the arguments in a function call.  Modelica doesn't have this kind of function today, but it could be provided by a library, or be added to the standard in the future.

If we still don't want sub-expression to be translatable, we should instead illustrate this with a similar example.

Reply:
In this case `"%name"` is preferable over `getInstanceName()`. Yes, I'm aware that it will be shorter - but to me that is a benefit.

Reply:
Sure, in many cases one would probably use `%name` instead, but it doesn't work for the purpose of this example.  The point of the example is about the general principle of finding the concatenated string literals, and I am trying to avoid the use of some imaginary under-defined `sprintf` function.

Would it be better if `getInstanceName()` was stored in a constant?
```
constant String myName = "instance named " + getInstanceName();
annotation(…, Text(string = "Hello, " +
    "the " + myName +
    " is an instance of %class."), …);

```